orte/test/mpi/singleton_client_server.c

```diff
@@ -81,12 +84,18 @@ int main(int argc, char *argv[])
 
   /* get the node name */
   {
-    int retval = gethostname(hostname, sizeof(hostname));
-    if(retval == -1)
+    temp_errno = errno;
```

Comment:
Because this `opal_gethostname()` function returns a char* rather than a return code, I am replicating the return code check that was here (that I removed) by zeroing out errno (which could be anything at this point) and then checking to see if there is a failure-type errno on return. If there isn't, I put it back to what it previously was because I want to avoid any unintended side effects. Remember, I'm touching a lot of code, I don't know what it's all doing, so I am trying to be cautious here.

I should probably only restore errno if it's zero on return from the function call, so that is a bug that I will fix. So thanks for bringing that to my attention. Also, because I'll be setting errno to ENOMEM if the malloc or realloc checks fail (per your suggestion), I will need to add that.

I realize this is not ideal but when it was decided that `opal_gethostname()` would return a char*, the die was cast for some clunkiness if we want to see if it succeeded or failed. One alternative is to just no longer check but that seems bad, especially in the case of a memory fail.

Reply:
I understand, and you're right, but gethostname() also had a return code. Anyway, it was an effort to be super-careful.